quickwit/quickwit-indexing/src/models/shard_positions.rs

```diff
@@ -78,29 +219,40 @@ impl PublishedShardPositions {
         } = &source_uid;
         let key = format!("{SHARD_POSITIONS_PREFIX}{index_uid}:{source_id}");
         let shard_positions_json = serde_json::to_string(&shard_positions).unwrap();
-        self.cluster_client
+        self.cluster
             .set_self_key_value(key, shard_positions_json)
             .await;
     }
 
+    fn publish_shard_updates_to_event_broker(&self, source_uid: SourceUid) {
+        let Some(shard_positions_map) = self.shard_positions_per_source.get(&source_uid) else {
+            return;
+        };
+        let shard_positions: Vec<(ShardId, Position)> = shard_positions_map
+            .iter()
+            .map(|(&shard_id, position)| (shard_id, position.clone()))
+            .collect();
+        self.event_broker.publish(ShardPositionsUpdate {
```

Comment:
At the moment, I don't see the value of sending complete updates. It's just more work for the ingester on truncate or the control plane on shard deletion to detect positions that have already been processed.